scapy/utils.py

```diff
@@ -1305,8 +1305,16 @@ def write(self, pkt):
             else:
                 pkt = pkt.__iter__()
             for p in pkt:
+
                 if not self.header_present:
                     self._write_header(p)
+
+                if self.linktype != conf.l2types.get(type(p), None):
```

Comment:
To avoid useless calls to `conf.l2types.get()` we could have a `set()` of types that are OK in this PCAP files. We could check if `p.__class__ not in self.l2_types_seen` first, only run this code block in that case, and add `self.l2_types_seen.add(p.__class__)`.
I'm not sure if I'm clear here... let me know what you think.

Reply:
I understood your point. However, I wonder if this useful as `conf.l2types` is a dictionary that contains ~20 elements. Adding a custom `set()` won't speed things up.

Reply:
Could you try out the performances impact of this PR ?
I'm wondering if we actually *need* to warn people against that. It's pretty obvious (to me)

Reply:
I did not identify any difference using

```
from scapy.all import *
import tempfile
import os

filename = tempfile.mktemp()
iter = (Ether() / IP(dst=inet_ntoa(struct.pack("!I", n))) / UDP() / DNS() for n in range(10000))
wrpcap(filename, iter)
os.remove(filename)
```